### ISSUE TITLE
Enum value told to not exist when clearly it does

### ISSUE THREAD

**Issue by Scalahansolo** [prisma/prisma#18240]
### Bug description

In my DB I create the following migration to add an enum to my cockroach db. 

`CREATE TYPE "ApiKeyReadWrite" AS ENUM ('NONE', 'READ', 'WRITE');`

In a later migration I added a column to a table that uses that type.

`ALTER TABLE "ApiKey" ADD COLUMN     "userAccess" "ApiKeyReadWrite" NOT NULL DEFAULT 'NONE';`

However.... at this point, my coworkers and now myself keep running into various errors that claim that type has not been created. For example... I just attempted to create a new migration with Prisma and was hit with the following error. It seems somehow to do with the `updatedAt` column on our `ApiKey` table, as we can get past any errors by applying these changes locally and running the migrations. 

<img width="681" alt="CleanShot 2023-03-06 at 21 18 15@2x" src="https://user-images.githubusercontent.com/4317253/223319652-3ac48d7a-ab2c-4e13-ac64-733e44d5b416.png">

However, this just seems like a short term and hacky solution. 

```
Error: P3006

Migration `20230214230236_api_key_user_access` failed to apply cleanly to the shadow database.
Error:
db error: ERROR: type "apikeyreadwrite" does not exist
   0: migration_core::state::DevDiagnostic
             at migration-engine/core/src/state.rs:269
```

### How to reproduce

See this minimal example to cause the issue. https://github.com/Scalahansolo/PrismaBug18240

Clone repo, apply migrations to a fresh DB. Add the new column to the Prisma schema, then attempt to generate a new migration with. 

On my setup I get hit with the following error.

<img width="1702" alt="CleanShot 2023-03-07 at 08 46 33@2x" src="https://user-images.githubusercontent.com/4317253/223473605-384b32a8-f1cc-47a6-a794-49f8fd71a8d0.png">

### Expected behavior

Migrations to "just work"

### Prisma information

The relevant pieces
```prisma
enum ApiKeyReadWrite {
  NONE
  READ
  WRITE
}

model ApiKey {
  ...
  createdTime          DateTime        @default(now())
  updatedTime         DateTime        @updatedAt
  taskAccess            ApiKeyReadWrite @default(NONE)
  commentAccess   ApiKeyReadWrite @default(NONE)
  projectAccess       ApiKeyReadWrite @default(NONE)
  workspaceAccess ApiKeyReadWrite @default(NONE)
  userAccess            ApiKeyReadWrite @default(NONE) # Added after the migration that creates this table.

  @@unique([userId, clientId])
}
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: MacOS arm64
- Database: Cockroach DB 22.2
- Node.js version: 16.17.1

### Prisma Version

```
prisma                  : 4.10.1
@prisma/client          : 4.10.1
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine aead147aa326ccb985dcfed5b065b4fdabd44b19 (at ../../node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli aead147aa326ccb985dcfed5b065b4fdabd44b19 (at ../../node_modules/@prisma/engines/migration-engine-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.10.1-1.80b351cc7c06d352abe81be19b8a89e9c6b7c110
Default Engines Hash    : aead147aa326ccb985dcfed5b065b4fdabd44b19
Studio                  : 0.481.0
```

**Comment by Scalahansolo** [comment]
Okay... I tracked this down.... Turns out this is actually a CockroachDB bug. CockroachDB is not preserving the casing when using Alter Table.

https://share.cleanshot.com/yF84Q8vz

**Comment by janpio** [comment]
That is unfortunately a bug in CockroachDB: https://github.com/cockroachdb/cockroach/issues/97493 Was fixed in https://github.com/cockroachdb/cockroach/pull/97514 - not sure if this has been included in a release yet.

**Comment by Scalahansolo** [comment]
Was meaning to update this yesterday. Like you mentioned, it was fixed. It was fixed in release 22.2.6. Closing this out.
